provisioning/virtual-machines/ansible/kubernetes/tasks/main.yaml

```diff
@@ -0,0 +1,96 @@
+---
+# Install kubernetes (not configure the cluster)
+
+  
+- name: Create docker daemon directory
+  file:
+    path: /etc/docker
+    state: directory
+
+- name: Create docker daemon file
+  copy:
+    src: files/daemon.json
+    dest: /etc/docker/daemon.json
+    owner: root
+    group: root
+    mode: '0644'
+  notify: Restart the docker service
+
+- name: Install resolvconf
+  apt:
+    name: "{{resolvconf}}"
+    state: latest
+    force_apt_get: yes
+  vars:
+    resolvconf:
+    - resolvconf
+
+- name: Resolvconf directory
+  file:
+    path: /etc/resolvconf/resolv.conf.d
+    state: directory
+
+- name: Set DNS to 8.8.8.8
+  copy:
+    content: "nameserver 8.8.8.8\n"
+    dest: /etc/resolvconf/resolv.conf.d/head
+
+- name: Enable resolvconf 
+  systemd:
+    enabled: yes
+    name: resolvconf
+
+- name: Restart resolvconf 
+  systemd:
+    state: restarted
+    name: resolvconf
+
+- name: Install kubernetes prerequisites
+  apt:
+    name: "{{ kubernetes_prerequisites }}"
+    state: present
+    force_apt_get: yes
+    update_cache: yes
+  vars:
+    kubernetes_prerequisites:
+    - apt-transport-https
+    - curl
+
+- name: Add kubernetes GPG-key
+  apt_key:
+    url: https://packages.cloud.google.com/apt/doc/apt-key.gpg
+    state: present
+
+- name: Add kubernetes repository
+  apt_repository:
+    repo: deb https://apt.kubernetes.io/ kubernetes-xenial main
+    state: present
+    update_cache: yes
+    filename: kubernetes.list
+
+- name: Install kubernetes
+  apt:
+    name: "{{ kubernetes_packages }}"
+    state: present
+  vars:
+    k8s_version: 1.19.0-00
+    kubernetes_packages: 
+    - "kubelet={{ k8s_version }}"
+    - "kubeadm={{ k8s_version }}"
+    - "kubectl={{ k8s_version }}"
```

Comment:
I would also pre-fetch a docker image via:
```
- name: pull an image
  docker_image:
    name: pacur/centos-7
    source: pull
```

Reply:
Great! Totally agree, this should strongly reduce the pressure on the network when creating the cluster

Reply:
Please add the proper comments in the yaml file to document this behavior. Document also that this has to be modified in case we move to a future K8s version.

Reply:
Wondering whether this may be replaced with something more general like:
```yaml
- name: ...
  command: kubeadm config images pull 
```

Reply:
I found out this was the recommended way by Kubernetes. So, I replaced the command with this new one.